lib/Parse/ParseExpr.cpp

```diff
@@ -3053,6 +3189,7 @@ bool Parser::ParseGenericFunctionExpression(ExprResult &Res) {
     return false;
   }
   else {
+    SmallVector<DeclRefExpr::GenericInstInfo::TypeArgument, 16> typeArgumentInfos;
```

Comment:
Instead of 16, could you just use 4?   It is unlikely someone will write a function with more than 1 or 2  type parameters.